docs/decisions/0007-release-process.md

```diff
@@ -0,0 +1,352 @@
+# Context and Problem Statement
+
+Releasing is an essential process for the `s3gw` process. Given the project is
+composed by various sub-projects, that need to be prepared, tested, and
+eventually released, the Release Process is not trivial.
+
+This document defines and agrees on the Release Process for the `s3gw` project,
+including the steps to be taken on its individual sub-projects, and results
+from several iterations defining the Release Process (previously kept in the
+[Release Captain ADR](/docs/decisions/0004-release-captain.md)). The previous
+document suffered significant changes since its inception, being modified for
+almost every release we have had; we are hoping the present document will
+benefit from more stability.
+
+## Definitions
+
+Throughout this document we will often refer to certain things or terms. Below
+we define what they mean.
+
+- Version: the state of a given project, at a specific point in time.
+
+- Release Candidate: the tentative set of deliverables from the various `s3gw`
+  sub-projects' repositories at a certain version. It may become the final Release.
+
+- Release: the set of deliverables from the various `s3gw` sub-projects'
+  repositories at a certain version, published, and accompanied by a
+  Release Statement.
+
+- Release Statement: a document associated with a given Release, detailing the
+  version being released, and a Changelog.
+
+- Changelog: a list of significant changes that merit being communicated to
+  stakeholders in a human-consumable format.
+
+- Release Pipeline: the set of automated workflows triggered on a specific
+  moment, resulting in a set of containers or artifacts to be released.
+
+- Release Branch: the state of a given project's development branch at a given
+  point in time, as a separate named branch in said project's git repository.
+
+- Backport: the act of applying a patch to a Release Branch originated in a
+  more recent Release Branch or the development branch.
```

Comment:
Does not "development branch" deserve to be capitalized as "Release Branch" ?

Reply:
I didn't define "development branch" because it seemed straightforward enough. For the same reason, I don't think it warrants being capitalized. :shrug: